src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -1311,30 +1317,31 @@ private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLen
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.values);
             }
 
+            Debug.Assert(values != null);
             if (values.Length == 0)
             {
                 return this;
             }
 
             if (values[0] != null)
             {
-                Append(values[0].ToString());
+                Append(values[0]!.ToString()); // TODO-NULLABLE: ! should not be required
```

Comment:
Please change the comment to `// TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/34644`